lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1487,12 +1487,82 @@ void NodeOp::getCanonicalizationPatterns(RewritePatternSet &results,
                                          MLIRContext *context) {
   results.insert<patterns::EmptyNode>(context);
 }
+// A register with constant reset and all connection to either itself or the
+// same constant, must be replaced by the constant.
+struct foldResetMux : public mlir::RewritePattern {
+  foldResetMux(MLIRContext *context)
+      : RewritePattern(RegResetOp::getOperationName(), 0, context) {}
+  LogicalResult matchAndRewrite(Operation *op,
+                                PatternRewriter &rewriter) const override {
+    auto reg = dyn_cast_or_null<RegResetOp>(op);
+    if (!reg)
+      return failure();
+    auto reset = dyn_cast_or_null<ConstantOp>(reg.resetValue().getDefiningOp());
+    auto resetSignal = reg.resetSignal();
+    // Check for a valid const reset.
+    // TODO: Shoud we add a check for resetSignal.isa<BlockArgument>()?
+    if (!reset || resetSignal.getType().isa<AsyncResetType>())
+      return failure();
+    // Find the one true connect, or bail
+    ConnectOp con;
+    for (Operation *user : reg->getUsers()) {
+      // If we see a partial connect or attach, just conservatively fail.
+      if (isa<PartialConnectOp>(user) || isa<AttachOp>(user))
```

Comment:
There are no constant analog values for this to be a register of which has an attach, yes?

Reply:
Right, removed the check.